src/core/evaluator.js

```diff
@@ -1371,16 +1401,15 @@ class PartialEvaluator {
         } else if (typeNum === PatternType.SHADING) {
           const shading = dict.get("Shading");
           const matrix = dict.getArray("Matrix");
-          pattern = Pattern.parseShading(
+          const objId = this.parsePattern({
+            keyObj: pattern,
             shading,
             matrix,
-            this.xref,
             resources,
-            this.handler,
-            this._pdfFunctionFactory,
-            localColorSpaceCache
-          );
-          operatorList.addOp(fn, pattern.getIR());
+            localColorSpaceCache,
+            localPatternCache,
+          });
+          operatorList.addOp(fn, ["Shading", objId]);
```

Comment:
To prevent any future issues, I believe that you also want to add `operatorList.addDependency(objId);` *before* this line since you're no longer sending the data inline here.

Reply:
The pattern should always be sent before the id is created, so a dependency pause shouldn't be needed. If it isn't sent first there will be an error on the main thread when we try to fetch the object (I did this accidentally).